orientdb/src/main/java/org/hibernate/datastore/ogm/orientdb/OrientDBDialect.java

```diff
@@ -63,123 +97,316 @@ public OrientDBDialect(OrientDBDatastoreProvider provider) {
+			log.debug( "removeTuple:Key:" + dbKeyName + " (" + dbKeyValue + "). remove: " + pstmt.executeUpdate() );
+		}
+		catch (SQLException e) {
+			log.error( "Can not remove entity", e );
+			throw new RuntimeException( e );
+		}
 	}
 
 	@Override
-	public Association getAssociation(AssociationKey key, AssociationContext associationContext) {
-		throw new UnsupportedOperationException( "Not supported yet." );
+	public Association getAssociation(AssociationKey associationKey, AssociationContext associationContext) {
+		log.debug( "getAssociation:AssociationKey:" + associationKey + "; AssociationContext" + associationContext );
+		EntityKey entityKey = associationKey.getEntityKey();
+		boolean existsPrimaryKey = EntityKeyUtil.existsPrimaryKeyInDB( provider.getConnection(), entityKey );
+		if ( !existsPrimaryKey ) {
+			// Entity now extists
+			return ASSOCIATION_NOT_FOUND;
+		}
+		Map<RowKey, Tuple> tuples = createAssociationMap( associationKey, associationContext );
+		return new Association( new OrientDBAssociationSnapshot( tuples ) );
+	}
+
+	private Map<RowKey, Tuple> createAssociationMap(AssociationKey associationKey, AssociationContext associationContext) {
+		List<Map<String, Object>> relationships = entityQueries.get( associationKey.getEntityKey().getMetadata() )
+				.findAssociation( provider.getConnection(), associationKey, associationContext );
+
+		Map<RowKey, Tuple> tuples = new HashMap<>();
+
+		for ( Map<String, Object> relationship : relationships ) {
+			OrientDBTupleAssociationSnapshot snapshot = new OrientDBTupleAssociationSnapshot( relationship, associationKey,
+					associationContext );
+			RowKey rowKey = convert( associationKey, snapshot );
+			tuples.put( rowKey, new Tuple( snapshot ) );
+		}
+		return tuples;
+	}
+
+	private RowKey convert(AssociationKey associationKey, OrientDBTupleAssociationSnapshot snapshot) {
+		String[] columnNames = associationKey.getMetadata().getRowKeyColumnNames();
+		Object[] values = new Object[columnNames.length];
+		log.debug( "convert: columnNames:" + columnNames.length );
+
+		for ( int i = 0; i < columnNames.length; i++ ) {
+			values[i] = snapshot.get( columnNames[i] );
+			log.debug( "convert: columnName:" + columnNames[i] + "; value:" + values[i] );
+		}
+		return new RowKey( columnNames, values );
 	}
 
 	@Override
 	public Association createAssociation(AssociationKey key, AssociationContext associationContext) {
-		throw new UnsupportedOperationException( "Not supported yet." );
+
+		log.debug( "createAssociation: AssociationKey:" + key + "; AssociationContext" + associationContext );
+		return new Association();
 	}
 
 	@Override
 	public void insertOrUpdateAssociation(AssociationKey key, Association association, AssociationContext associationContext) {
-		throw new UnsupportedOperationException( "Not supported yet." );
+		log.debug( "insertOrUpdateAssociation: AssociationKey:" + key + "; AssociationContext:" + associationContext + "; association:" + association );
```

Comment:
I would prefere to have an exception saying that this method is not implemented than a log